Sources/Safetensors/Safetensors.swift

```diff
@@ -231,34 +47,26 @@ public enum Safetensors {
         let data = try Data(contentsOf: url, options: .mappedIfSafe)
         return try decode(data)
     }
-
+    
     ///  Decode Data object to ParsedSafetensors object.
     /// - Parameter data: `Data` object containing the encoded data
     /// - Returns: `ParsedSafetensors` object containing the decoded data
     public static func decode(_ data: Data) throws -> ParsedSafetensors {
-        guard data.count >= 8 else {
+        guard data.count >= MemoryLayout<Int>.size else {
             throw SafetensorsError.invalidHeaderSize
         }
-        let (headerOffset, headerData) = try data[0..<8].withUnsafeBytes {
-            (ptr: UnsafeRawBufferPointer) in
-            let headerSize = ptr.load(as: Int.self)
-            guard data.count >= 8 + headerSize else {
-                throw SafetensorsError.invalidHeaderData
-            }
-            let headerData = data[8..<8 + headerSize]
-            let decoder = JSONDecoder()
-            decoder.keyDecodingStrategy = .convertFromSnakeCase
-            let header = try decoder.decode([String: HeaderElement].self, from: headerData)
-            return (headerSize + 8, header)
-        }
-        try validate(header: headerData, dataCount: data.count - headerOffset)
+
+        let result = try HeaderDecoder().decode(data)
```

Comment:
I think there is no need to instantiate a struct just to decode some data, this can be just a free function I think